lib/web/app/middleware.go

```diff
@@ -87,7 +87,12 @@ func (h *Handler) redirectToLauncher(w http.ResponseWriter, r *http.Request, p l
 		return trace.Wrap(err)
 	}
 
-	urlString := makeAppRedirectURL(r, h.c.WebPublicAddr, addr.Host(), p)
+	var proxyPublicAddrs []string
+	for _, proxyAddr := range h.c.ProxyPublicAddrs {
+		proxyPublicAddrs = append(proxyPublicAddrs, proxyAddr.String())
+	}
+	proxyDNSName := utils.FindMatchingProxyDNS(r.Host, proxyPublicAddrs)
+	urlString := makeAppRedirectURL(r, proxyDNSName, addr.Host(), p)
```

Comment:
Is this the redirect that is called for both regular user auth and per session MFA?

Reply:
this is only for app auth redirect launch. not MFA redirect